README.md

```diff
@@ -1,101 +1,272 @@
+```json
+{
+  "providers": {
+    "github": {
+      "details": {
+        "dev": {
+          "clientId": "1440dd4c1d1c4c0fa124",
+          "clientSecret": "0723a2b5bfef27efc8b2d26d837ead239fa0b0e6",
+          "redirectUrl": "http://localhost:3000/redirect",
+          "scope": "repo, user:email, read:user, read:org"
+        },
+        "deploy": {
+          "clientId": "b4337a9c1ad66984c060",
+          "clientSecret": "1b05648d36ca92d63766e32e8e9a7d740b214816",
+          "redirectUrl": "https://snek.at/redirect",
+          "scope": "repo, user:email, read:user, read:org"
+        }
+      },
+      "urls": {
+        "authorizeUrl": "https://github.com/login/oauth/authorize?redirect_uri=",
+        "accessTokenUrl": "https://github.com/login/oauth/access_token?code=",
+        "usernameUrl": "https://api.github.com/user"
+      }
+    }
+  },
+  "proxyUrl": "https://cors.snek.at/",
+  "devUrl": "http://localhost:3000"
+}
+```
+<mark>Currently, a solution to adapt the config file more easily is under development.</mark>
 
-**Schett Nico**
+#### Implementation
+##### Redirect page
+Put this script inside your redirect page to obtain the access token from the OAuth2 request. 
+```javascript
+/*
+ * The redirect script:
+ * Get name of window which was set by the parent to be the unique
+ * request key
+ */
+const requestKey = window.name;
+// Update corresponding entry with the redirected url which should
+// either contain the access token or the reason for failure in the query
+// parameter / hash
+localStorage.setItem(requestKey, window.location.href);
+window.close();
+```
+##### Requesting access token
+```javascript
+/* Import GithubProvider and the RSA */
+import RSA, { GithubProvider } from "snek-intel/utils/oauth";
 
-- <https://twitter.com/sxct4>
-- <https://github.com/schettn>
+/* Gets the username and accessToken */
+const openGitHubOAuth = () => {
+    return await RSA.acquireTokenAsync(GithubProvider)
+};
 
-## [](#thanks)Thanks
+/* Example on how to call the GitHub OAuth function */
+const { data } = openGitHubOAuth();
+
+console.log(data.username, data.accessToken)
+```
+## [](#contributing)Contributing
+![GitHub last commit](https://img.shields.io/github/last-commit/snek-at/intel) ![GitHub issues](https://img.shields.io/github/issues-raw/snek-at/intel) ![GitHub closed issues](https://img.shields.io/github/issues-closed-raw/snek-at/intel?color=green)
 
-We do not have any external contributors yet, but if you want your name to be here, feel free
-to [contribute to our project](#contributing).
+Please read through our [contributing guidelines](https://github.com/snek-at/front/blob/master/CONTRIBUTING.md). Included are directions for opening issues, coding standards, and notes on development.
 
-## [](#copyright-and-license)Copyright and license
+All HTML and CSS should conform to the [Code Guide](https://github.com/mdo/code-guide), maintained by [Mark Otto](https://github.com/mdo).
```

Comment:
We should replace this with a reference to TONIC.
In tonic there will be a reference to Mark Otto